qiskit_machine_learning/kernels/quantum_kernel.py

```diff
@@ -90,6 +114,94 @@ def quantum_instance(
+        """Sets the user parameters"""
+        self._user_param_binds = {user_params[i]: user_params[i] for i, _ in enumerate(user_params)}
+
+        self._user_parameters = user_params
+
+    def assign_user_parameters(
+        self, values: Union[Mapping[Parameter, ParameterValueType], Sequence[ParameterValueType]]
+    ) -> None:
+        """
+        Assign user parameters in the QuantumKernel feature map.
+
+        Args:
+            values (dict or iterable): Either a dictionary or iterable specifying the new
+                parameter values. If a dict, it specifies the mapping from ``current_parameter`` to
+                ``new_parameter``, where ``new_parameter`` can be a parameter object or a
+                numeric value. If an iterable, the elements are assigned to the existing parameters
+                in the order of ``QuantumKernel.user_parameters``.
+
+        Raises:
+            ValueError: Incompatible number of user parameters and values
+        """
+        if self._user_parameters is None:
+            raise ValueError(
+                """
+                The number of parameter values ({len(values)}) does not
+                match the number of user parameters tracked by the QuantumKernel
+                (None).
+                """
+            )
+
+        if isinstance(values, dict):
+            unknown_parameters = list(set(values.keys()) - set(self._user_parameters))
+            if len(unknown_parameters) > 0:
+                raise ValueError(
+                    f"Cannot bind parameters ({unknown_parameters}) not tracked by the quantum kernel."
+                )
+            param_binds = values
+        else:
+            if len(values) != len(self._user_parameters):
+                raise ValueError(
+                    f"""
+                The number of parameter values ({len(values)}) does not
+                match the number of user parameters tracked by the QuantumKernel
+                ({len(self._user_parameters)}).
+                """
+                )
+            param_binds = {param: values[i] for i, param in enumerate(self._user_parameters)}
+
+        if self._user_param_binds is None:
+            self._user_param_binds = param_binds
+        else:
+            self._user_param_binds.update(param_binds)
+        self._feature_map = self._unbound_feature_map.assign_parameters(self._user_param_binds)
+
+    @property
+    def user_param_binds(self) -> Mapping[Parameter, float]:
+        """Return a copy of the current user parameter mappings for the feature map circuit."""
+        return copy.deepcopy(self._user_param_binds)
+
+    def bind_user_parameters(self, values: Sequence[float]) -> None:
+        """
+        Alternate function signature for assign_user_parameters
+
+        Args:
+            values (iterable): [value1, value2, ...]
+        """
+        self.assign_user_parameters(values)
+
+    def unbound_user_parameters(self) -> List[Parameter]:
```

Comment:
I guess this could be a property if it was desired. Should the docstring say 'Returns a list of the unbound....' - yield is something particular in Python